Fixing sorting in Ansible Tower Configuration Manager

pkomanek opened the pull request:
Purpose or Intent
-----------------
Sorting broken by Total Configured Systems. There was missing virtual
column total_configured_systems. Solved by adding virtual column and
taking total_configured_systems from related ext_management_system.

Links
-----
https://bugzilla.redhat.com/show_bug.cgi?id=1350308

cc @gmcculloug 
Thanks @lpichler

gmcculloug commented:
@bdunne Please review

bdunne commented on app/models/ems_folder.rb:
```diff
@@ -19,6 +19,13 @@ class EmsFolder < ApplicationRecord
+  # Methods for virtual columns
+  #
+
+  delegate :total_configured_systems, :to => :ext_management_system
```
Based on the decsription, I believe you are here, correct?
![image](https://cloud.githubusercontent.com/assets/651659/16620688/6694f2ee-4361-11e6-8c5b-90857a0e7c69.png)

I see the same error when trying to sort by Total Configured Systems.

However, delegating `total_configured_systems` to `ext_management_system` will get you the total number of Configured System for the entire provider which would not be the correct number.  Instead, you want the total for the Inventory Group.  I think your method should be:
```ruby
def total_configured_systems
  configured_systems.count
end
```

This also makes me question how the table was build the first time since this method doesn't exist...  There is probably another location that would benefit from using this method.

Also, I believe this logic should live in the InventoryGroup model, since it only applies there.

pkomanek commented:
I changed the db where the total_configured_systems column is searched from general model EmsFolder to more specialized model InventoryGroup. I also added this virtual column to InventoryGroup.

bdunne commented:
:+1: LGTM

gmcculloug commented:
@pkomanek I think it is ok remove the WIP label now, correct?

lpichler commented:
@miq-bot remove_label wip

chessbyte commented:
Darga backport details:
```sh
commit e73cf685012e52d5728380bd6444f1240d1d5391
Author: Greg McCullough <gmccullo@redhat.com>
Date:   Tue Jul 12 10:09:39 2016 -0400

    Merge pull request #9594 from pkomanek/fix_sorting_in_inventory_group_in_ansible_tower_provider
    
    Fixing sorting in Ansible Tower Configuration Manager
    (cherry picked from commit f8c72565705789b9692f8320c9a6aab2777988cb)
```
